2614 geometry refactor projection angles handling

hussam-stfc opened the pull request:
<!-- Close or ref the associated ticket, e.g.  -->
## Issue Closes #<ISSUE>

### Description

The following changes have been made:
- Refactored `ProjectionAngles` logic in `Geometry` and `ImageStack`. `Geometry` is now the sole source of truth for angle data.
- Added guards against `None` for `ProjectionAngles` at point of use
- Updated some reconstruction tests to use more accurate 3D-array test data.
- Improved method docstrings in reconstruction window and I/O testing code.

<!-- Add a description of the changes made. -->

### Developer Testing 

<!-- Describe the tests that were used to verify your changes -->
- I have verified unit tests pass locally: `python -m pytest -vs`
- I have manually verified that reconstruction continues to work as expected

### Acceptance Criteria and Reviewer Testing
 - [ ] Unit tests pass locally: python -m pytest -vs
 - [ ]  Verify that reconstruction performs as expected (recon results match previous commits)

### Documentation and Additional Notes

<!-- Please un-comment any of the below checkboxes applicable for your PR. This could be updated release notes, sphinx documentation, or screenshot tests.
Please also add any additional notes that may be helpful to the reviewer here including screenshots if necessary -->

- [x] Release Notes have been updated

<!--
- [ ] Sphinx documentation has been updated
- [ ] Screenshot tests have been updated
  - [ ] **Before merge for developer:** Resolve the change on applitools by, creating a new baseline for test and verify that the tests pass.
  - [ ] **After merge for reviewer:** Go to [Applitools compare and merge page](https://eyes.applitools.com/app/merge/), select all changes and merge dev branch to default branch
  - [ ] **After merge for reviewer:** Verify that other PR's screenshot tests do not start to fail (See [Applitools baselines](https://eyes.applitools.com/app/baselines) for more info) -->

hussam-stfc committed:
Update tests in recon/model_test.py to use more accurate 3D array test data

These tests pass due to ImageStack's width and height properties never being called with this test data. However, the 1D arrays previously used are not accurate reflections of real data

hussam-stfc committed:
Add guards against ProjectionAngles data being None

hussam-stfc committed:
Remove real_projection_angles() method and update all references to use projection_angles()

With the new model of storing angles in Geometry rather than ImageStack, there is no longer a need for both methods

hussam-stfc committed:
Update Geometry to be the primary store for angle data

New test helper methods have been added, and relevant tests have been updated

hussam-stfc committed:
Improve method docstrings in recon window and io testing

hussam-stfc committed:
Add release note (#2614)

hussam-stfc committed:
Update system tests to account for ImageStack no longer holding Geometry

samtygier-stfc commented on mantidimaging/gui/test/gui_system_loading_test.py:
```diff
@@ -75,29 +75,43 @@ def _get_log_angle(self, log_path):
         self.main_window.actionSampleLoadLog.trigger()
-
         self.assertIn("log_file", sample.metadata)
         self.assertEqual(sample.metadata['log_file'], str(log_path))
```
Could also check the angles here, as they will now have been loaded from the log.

samtygier-stfc reviewed (changes requested):
Looking good. A few changes, but they are mostly small things.

hussam-stfc committed:
Restore recovery of original cor/tilt in astra_recon's find_cor() so that they get reset after the new cor is found

hussam-stfc committed:
Update projection angles test to check if angles are none with no log file

hussam-stfc committed:
Remove redundant imagestack generator helper function from image_stack_test.py

samtygier-stfc committed:
Remove some checks from test_get_projection_angles_no_logfile

They parts are check by other tests

samtygier-stfc committed:
Reorder Dataset creation in_load_strict_data_set() to avoid visual change

If the name is set before the dataset is created, then it gets used as the dataset name

samtygier-stfc committed:
add_alternative_180_if_required() keep old behaviour of using middle projection in case of no angles

This will likely be remove soon

samtygier-stfc reviewed (approved):
I have fixed the last few comments and small issue for stacks without angles.

Tested with several datasets.
